src/safeposix/syscalls/fs_calls.rs

```diff
@@ -3058,91 +3146,181 @@ impl Cage {
+    /// [close(2)](https://man7.org/linux/man-pages/man2/close.2.html)
     pub fn close_syscall(&self, fd: i32) -> i32 {
-        //check that the fd is valid
+        // Call the helper function which validates the result and deletes
+        // reference of the file descriptor.
         return Self::_close_helper(self, fd);
     }
 
+    /// ## ----------------- CLOSE HELPER INNER ------------------
+    /// ### Description
+    ///
+    /// The `_close_helper_inner` function manages the cleanup and closing of
+    /// various types of file descriptors (files, directories, pipes, sockets)
+    /// by decrementing reference counts and removing inodes if necessary.
+    /// ### Function Arguments
+    ///
+    /// The function receives one argument:
+    /// * `fd` - This argument refers to the file descriptor to be closed.
+    ///
+    /// ### Returns
+    ///
+    /// Upon successful completion of this call, it returns 0. Otherwise, it
+    /// returns errors set with the appropriate error code.
+    ///
+    /// ### Errors
+    ///
+    /// * EBADF - The given file descriptor is invalid
+    /// * ENOEXEC - The Non-regular type file (dir/chardev) in file object table
+    ///
+    /// ### Panics
+    ///
+    /// * If the provided file descriptor is out of bounds, causing unwrap() to
+    ///   panic
+    /// * When the file type filedescriptor contains a Socket as an inode.
+    ///
+    /// For more detailed description of all the commands and return values, see
+    /// [close(2)](https://man7.org/linux/man-pages/man2/close.2.html)
     pub fn _close_helper_inner(&self, fd: i32) -> i32 {
+        //BUG
+        //if the provided file descriptor is out of bounds, get_filedescriptor returns
+        // Err(), unwrapping on which  produces a 'panic!'
+        //otherwise, file descriptor table entry is stored in 'checkedfd'
         let checkedfd = self.get_filedescriptor(fd).unwrap();
         let mut unlocked_fd = checkedfd.write();
         if let Some(filedesc_enum) = &mut *unlocked_fd {
-            //Decide how to proceed depending on the fd type.
-            //First we check in the file descriptor to handle sockets (no-op), sockets
-            // (clean the socket), and pipes (clean the pipe), and if it is a
-            // normal file descriptor we decrement the refcount to reflect
+            // We decide, how to proceed depending on the fd type.
+            // First we check in the file descriptor to handle stream / epoll (no-op),
+            // sockets (clean the socket), and pipes (clean the pipe), and if it is a
+            // normal file descriptor we decrement the reference count to reflect
             // one less reference to the file.
             match filedesc_enum {
                 //if we are a socket, we dont change disk metadata
-                Stream(_) => {}
-                Epoll(_) => {} //Epoll closing not implemented yet
+                Stream(_) => {} // Stream closing not supported
```

Comment:
I think theres some misinterpretation through out here. Its not that closing isnt supported, its just that there is no additional cleanup to do. Of course all file descriptors can be closed.

Reply:
Oh, I got it. Thanks for pointing this out, I have updated the comment and checked for other cases as well.